File: internal/model/playlist.go
package model

import "time"

// Playlist represents a user-created playlist.
type Playlist struct {
	ID          string    `json:"id"`
	UserID      string    `json:"userId"`
	Name        string    `json:"name"`
	Description string    `json:"description,omitempty"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

// PlaylistTrack represents the association between a playlist and a track.
type PlaylistTrack struct {
	PlaylistID string    `json:"playlistId"`
	TrackID    string    `json:"trackId"`
	Position   int       `json:"position"`
	AddedAt    time.Time `json:"addedAt"`
}

// Favorite represents a user's favorited item (track, album, or artist).
type Favorite struct {
	UserID    string    `json:"userId"`
	ItemType  string    `json:"itemType"` // "track", "album", "artist"
	ItemID    string    `json:"itemId"`
	CreatedAt time.Time `json:"createdAt"`
}

File: ios/Sonus/Models/Playlist.swift
import Foundation

/// A user-created playlist.
struct Playlist: Codable, Identifiable, Hashable, Sendable {
    let id: String
    let userId: String
    let name: String
    let description: String?
    let createdAt: Date
    let updatedAt: Date
}

/// Association between a playlist and a track.
struct PlaylistTrack: Codable, Hashable, Sendable {
    let playlistId: String
    let trackId: String
    let position: Int
    let addedAt: Date
}

/// A user's favorited item (track, album, or artist).
struct Favorite: Codable, Hashable, Sendable {
    let userId: String
    let itemType: String
    let itemId: String
    let createdAt: Date
}

/// A single listening history event.
struct ListeningHistory: Codable, Identifiable, Hashable, Sendable {
    let id: Int64
    let userId: String
    let trackId: String
    let playedAt: Date
    let durationSeconds: Int?
}
